src/AutoRest.CSharp/Common/Output/Models/Requests/QueryParameter.cs

```diff
@@ -1,9 +1,24 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using System;
+using System.Collections.Generic;
 using AutoRest.CSharp.Output.Models.Serialization;
 
 namespace AutoRest.CSharp.Output.Models.Requests
 {
-    internal record QueryParameter(string Name, ReferenceOrConstant Value, string? Delimiter, bool Escape, SerializationFormat SerializationFormat, bool Explode);
+    internal record QueryParameter(string Name, ReferenceOrConstant Value, string? Delimiter, bool Escape, SerializationFormat SerializationFormat, bool Explode)
+    {
+        // see https://github.com/Azure/autorest/blob/ca75e7a8a3caf6ce649fafdded832449c6250efb/packages/extensions/modelerfour/src/modeler/interpretations.ts#L77
+        private static readonly HashSet<string> APIVersionParameterNames = new(StringComparer.OrdinalIgnoreCase)
+        {
+            "api-version",
+            "apiversion",
+            "x-ms-api-version",
+            "x-ms-version"
+        };
+
+        public bool IsApiVersion => APIVersionParameterNames.Contains(Name);
```

Comment:
modelerfour already has this information. This means, in the origin of this record (`RequestParameter`), there is a property `Origin`. If the parameter is api-version, it will has a value of `synthesized/api-version` or something - and we should decide whether it is an api-version parameter by that.
See here: https://github.com/Azure/autorest.csharp/blob/dcb40205c60154c8c2ef1dcdee5d2bd11ae853f4/src/AutoRest.CSharp/Common/Input/CodeModelConverter.cs#L134
In this way, we no longer need to hard code this in our code.